fixed health floating point error, fungal id vs name confusion

teh60 opened the pull request:
- extra health perk multiplies max HP by 1.5 thus taking it off integer values, so added a floor divide to fix BigInt handling
- fungal shifts were confusing with multiple id's mapping to same translated name, added distinction hover and changed lua mod to report both id and translated name

Commit:
fixed health floating point error, fungal id vs name confusion

Commit:
bumped version

myndzi committed:
Update package.json

Commit:
improved material key to material name conversion

Commit:
added support for new spent extra life perk

Commit:
improved map preview system and fungal system
- player is now shown as a star ontop of the map image
- heaven/hell loops are now properly tracked
- added fungal timer/ready to shift info line
- refactored map preview calculation/coordinate handling
- added extra error report to try and track down json parsing issue

Commit:
bumped version

Commit:
removed part of map disclaimer note

Commit:
fixed express vulnerability and fixed noita lua mod bug with fungal info (requires new release)

Commit:
added support for online wand simulator
- new "Tinker" button on wands loads the wand stats/spells into the wand simulator
- added wand speed multiplier to wand stats
- added missing wand sprites
- fixed style issue on world seed readout
- fixed fungal shift info error in mongoDB vs live update data

Commit:
added compatability with conga's spell lock mod and updated spell info
- streamer-wands mod detects if spell lock is enabled and will grey out spells on the progress table if they are locked out
- fixed explosive damage and radius for a few spells
- forgot a const on zoom variable in osd function (OCD strikes again)

Commit:
fixed fungal shifting errors and inaccuracies
- fixed fungal time remaining counting down during first 5 minutes when no shift has occured during that time
- fixed an edgecase where a functional chain was being detected as a broken chain
- fixed previous shifts being mis-cancelled out when a material shifts into itself

Commit:
reworked fungal shift info
- new codebase for fungal shift calculation
- reworked fungal shift material database/LUA ws method
- new format for displaying shifts
- see PR comment for more details

Commit:
bump rc version

Commit:
fixed fungal data only updating on page refresh

Commit:
improved "Map and Game info" display
- added readout for what gamemode/world generation mod is active
- map preview now supports nightmare, NG+ and some noita gamemode maps and world generation mods
- restructured worldComponent and created mapComponent to better organize and isolate front-end functionality

Commit:
mods and map info displays were default open instead of closed

Commit:
updated map info
- due to noita update map tile links needed to be updated
- made the displayed seed icnrease for NG+ loops for accurate fungal shift data
    - also added a tooltip to notify of this

Commit:
reworked map info to pull from map.runfast.stream's json data to eliminate manual data scraping

myndzi commented on public/main.js:
```diff
@@ -538,24 +538,18 @@ const mapComp = Vue.component('map-comp', {
-                placement: 'right',
-                modifiers: [{ name: 'offset', options: { offset: [0, 5] } }],
+        fetch("https://map.runfast.stream/js/tilesources.json")
+            .then(res => res.json())
```
A good habit here is something like:

```suggestion
            .then(res => (res.ok ? res.json() : Promise.reject(`HTTP ${res.status}: ${res.statusText}`)))
```

Fetch promises can resolve (successfully complete) with a failure (status in 4xx-5xx) or an unexpected result (status in the 1xx or 3xx range), so you've gotta be sure to catch both "successful failures" and failures

myndzi commented on public/main.js:
```diff
@@ -538,24 +538,18 @@ const mapComp = Vue.component('map-comp', {
+            .then(data => {
+                this.mapData = data
+                this.loaded = true
             })
```
Do something on failure, too with `.catch(err => ....)`, even if it's just displaying some text like "failed to fetch map data"

myndzi commented on public/main.js:
```diff
@@ -614,12 +609,20 @@ const mapComp = Vue.component('map-comp', {
+            // get map URL and topleft offset
+            const mapValues = this.mapData[mapName]
+            const map = {
+                urls: mapValues.map((x) => `${x.url.slice(0, -4)}_files/`),
```
You could use something like `${x.url.replace(/\.dzi$/, '')}` to make the intent clearer here. In real code I'd be looking for validation of the (unknown/untrusted) input data, but the value of adding that here is minimal.

Commit:
added fetch resolve checking, fetch error handling
also cleaned up map data string-manipulation code

Commit:
fixed noitool seed link

Commit:
improved TINKER link
- added tooltip with old tinker link, and description about both
- made both tinker links blank out incompatible spells

Commit:
fixed stats error due to missing method

myndzi reviewed (approved):
Between repeated live tests on the dev domain and various other convos, I guess it's time to pull the trigger 😅
